controllers/lookup/k8s_client.go

```diff
@@ -0,0 +1,160 @@
+package lookup
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/datainfrahq/druid-operator/apis/druid/v1alpha1"
+	internalhttp "github.com/datainfrahq/druid-operator/pkg/http"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"net/http"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type K8sClient struct {
+	client client.Client
+}
+
+func NewK8sClient(client client.Client) K8sClient {
+	return K8sClient{
+		client: client,
+	}
+}
+
+func (c *K8sClient) FindLookups(ctx context.Context, reports map[types.NamespacedName]Report) (map[ClusterKey]map[LookupKey]Spec, error) {
+	specs := &v1alpha1.DruidLookupList{}
+	if err := c.client.List(ctx, specs); err != nil {
+		return nil, err
+	}
+
+	lookupSpecsPerCluster := make(map[ClusterKey]map[LookupKey]Spec)
+	for _, spec := range specs.Items {
+		clusterKey := ClusterKey{
```

Comment:
You don't need to create a new struct for that. This can be stored in NamespacedName struct.

Reply:
Replaced `ClusterKey` with `NamespacedName` in 7c6e6e3.